Documentation/coding-guidelines/project-guidelines.md

```diff
@@ -72,103 +106,40 @@ Project configurations that are unique for a few different target frameworks and
-    </Project>
-    <Project Include="System.Linq.Expressions.csproj">
-      <OSGroup>Windows_NT</OSGroup>
-      <TargetGroup>uap101aot</TargetGroup>
-    </Project>
-  </ItemGroup>
-  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.traversal.targets))\dir.traversal.targets" />
-</Project>
+src\<Library Name>\src - Contains the source code for the library.
+src\<Library Name>\ref - Contains any reference assembly projects for the library
+src\<Library Name>\pkg - Contains package projects for the library.
+src\<Library Name>\tests - Contains the test code for a library
 ```
+##ref
+Reference assemblies are required for any library that has more than one implementation or uses a facade. A reference assembly is a surface-area-only assembly that represents the public API of the library. To generate a reference assembly source file you can use the [GenAPI tool](https://www.nuget.org/packages/Microsoft.DotNet.BuildTools.GenAPI). If a library is a pure portable library with a single implementation it need not use a reference assembly at all.
 
-###Tests project .builds files
+In the ref directory for the library there should be at most **one** `.csproj` that contains the latest API for the reference assembly for the library.
 
-The tests .builds files are very similar to the regular ones, except that they usually tend to pass in one extra property as metadata: `TestTFMs`. The purpose for this extra metadata property, is to show which TFMs are supported by the test projects build configuration. When doing a full build, a TestTFM will be specified (if not specified netcoreapp1.1 will be used as default), and the build will look into all of these test .builds files to try and find which configurations support testing in that TestTFM, and then start running the tests for those configurations.
+There are two types of Reference assembly projects:
 
-####How to know which TestTFMs does a project support
-`TargetGroup` and `TestTFM` are closely  tied together, given that `TargetGroup` selects the TFM (the surface area) that the test assembly will use, and the `TestTFM` is where tests will actually run on. Here is a small cheat sheet of which `TestTFMs` you should add to the builds file given a `TargetGroup`:
+1. Libraries that are contain APIs in netstandard
+ - `BuildConfigurations` should contain non-netstandard configurations for the platforms they support.
+ - Should use relative path `<ProjectReference>` to the dependencies it has. Those depedencies should only be libraries with similar build configurations and be part of netstandard.
+2. Libraries that are built on top of netstandard
+ - `BuildConfigurations` should contain only netstandard configurations.
+ - Should contain `<Reference Include='netstandard'>`
+ - Anything outside of netstandard should use a relative path `<ProjectReference>` to its dependencies it has. Those depdencies should only be libraries that are built against netstandard as well.
```

Comment:
I'm thinking of a potential case where someone builds against netstandard but also needs to reference something else built against netstandard. I'm not sure if we have any such cases but I can imagine we will.